python/tk_desktop/desktop_window.py

```diff
@@ -799,8 +800,9 @@ def __launch_app_proxy_for_project(self, project, pipeline_configuration_id=None
                 "proxy_auth": server_auth,
             },
         }
-        (_, pickle_data_file) = tempfile.mkstemp(suffix='.pkl')
+        (fd, pickle_data_file) = tempfile.mkstemp(suffix='.pkl')
         pickle.dump(desktop_data, open(pickle_data_file, "wb"))
```

Comment:
Well, in this branch you are worried that file descriptors will be leaked and this is a potential leak, so it would make sense to clean that up as well. :)